src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -199,6 +199,46 @@ def visit(self, visit_node: Callable[[DependencyGraph], bool | None], visited: s
     def new_dependency_graph_context(self):
         return DependencyGraphContext(parent=self, path_lookup=self._path_lookup, session_state=self._session_state)
 
+    def _compute_route(self, root: Path, leaf: Path, visited: set[Path]) -> list[Dependency]:
+        # leaf is not on the route so handle case where it doesn't inherit context
+        maybe = self.locate_dependency(leaf)
+        assert maybe.graph  # assume caller knows what they're doing
+        if not maybe.graph.dependency.inherits_context:
+            return [maybe.graph.dependency]
+        # handle generic case
+        route = self._do_compute_route(root, leaf, visited)
+        return self._trim_route(route)
+
+    def _do_compute_route(self, root: Path, leaf: Path, visited: set[Path]) -> list[Dependency]:
+        maybe = self.locate_dependency(root)
+        assert maybe.graph  # assume caller knows what they're doing
```

Comment:
don't assume things like that. return a dependency problem if this occurs.

Reply:
done